### ISSUE TITLE
LexerSkipAction::INSTANCE and others are nullptr

### ISSUE THREAD

**Issue by Nic30** [DanMcLaughlin/antlr4#15]
I do not know why or how, but 
```
LexerMoreAction::INSTANCE
LexerPopModeAction::INSTANCE
LexerSkipAction::INSTANCE
```
are nullptr when 
`Ref<LexerAction> ATNDeserializer::lexerActionFactory(LexerActionType type, int data1, int data2)`
is called, it seems like bug in compiler or misuse of shared_ptr

( 

g++ (Ubuntu 5.3.1-14ubuntu2.1) 5.3.1 20160413  (default in Ubuntu 16.04)
It seems that shared pointer is destroyed or initializer is never called.

)

Which compiler are you using? 

This causes one of bugs mentioned in #12

**Comment by Nic30** [comment]
Constructors of LexerSkipAction and others are never called but they always should be, for example in 
`const Ref<LexerSkipAction> LexerSkipAction::INSTANCE { new LexerSkipAction() };`

**Comment by Nic30** [comment]
I have tried g++-6 (Ubuntu 6.1.1-2ubuntu12~16.04) 6.1.1 20160510,
but results are same.

**Comment by Nic30** [comment]
It seems to be same problem as described there http://stackoverflow.com/questions/19147588/shared-pointers-to-a-singleton-do-not-recognize-each-other

**Comment by nuuman** [comment]
Nice work @Nic30

**Comment by mike-lischke** [comment]
Looks like there is a problem with static initialization. I'll look after this when I return from vacation.

**Comment by Nic30** [comment]
@mike-lischke great, I have send you pull request (from https://github.com/Nic30/antlr4).
It is working for me.

**Comment by mike-lischke** [comment]
Fixed after pull request #16 has been merged.
